### PR TITLE
Fix registry delete

### LINKED ISSUE #520: deleting registry in settings does not remove the registry
### Component

server, web-ui

### Describe the bug

reproducible steps

1. add two private registries to a repo
2. remove a registry
3. notice that two registries still remain

expected
1. only one registry is left

### System Info

```shell
next-4891f104
```

### Additional context

_No response_

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/woodpecker-ci/woodpecker/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't [already an issue](https://github.com/woodpecker-ci/woodpecker/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A join our [Discord Chat Server](https://discord.gg/fcMQqSMXJy).

### PR DESCRIPTION
**Pull request by 6543**
* [x] Make make it possible to delete `"§4fda`
* [x] reject non urls for registry-address
* [x] fix #520

### CODE REVIEW

File: web/src/components/repo/settings/RegistriesTab.vue [hunk 1]
```diff
@@ -35,6 +35,7 @@
     <div v-else class="space-y-4">
       <form @submit.prevent="createRegistry">
         <InputField label="Address">
+          <!-- TODO: check input field Address is a valid address -->
```
Comment (6543) [inline web/src/components/repo/settings/RegistriesTab.vue:38]: @anbraten do you have a nice way to do this in frontend?

Comment (anbraten) [reply web/src/components/repo/settings/RegistriesTab.vue:38]: We can add something like [vuelidate](https://vuelidate-next.netlify.app/) in the future. For now it should show you the error from the backend as a toast notification.

Comment (6543) [reply web/src/components/repo/settings/RegistriesTab.vue:38]: ok I'll leave the TODO and we can do this later -> #533

File: web/src/components/repo/settings/RegistriesTab.vue [hunk 2]
```diff
@@ -115,7 +116,8 @@ export default defineComponent({
         throw new Error("Unexpected: Can't load repo");
       }
 
-      await apiClient.deleteRegistry(repo.value.owner, repo.value.name, _registry.address);
+      const registryAddress = encodeURI(_registry.address).replaceAll('/', '%2F');
```
Comment (6543) [inline web/src/components/repo/settings/RegistriesTab.vue:119]:
```suggestion
      const registryAddress = encodeURIComponent(_registry.address);
```

**Comment by 6543** [comment]
https://github.com/gin-gonic/gin/pull/810
https://github.com/gin-gonic/gin/issues/1528

**Review by anbraten — APPROVED** [review]

**Pull request merged**